### ISSUE TITLE
SC3040 has no documentation at all

### ISSUE THREAD

**Issue by rugk** [koalaman/shellcheck#2308]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC3040
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set -euxo pipefail
```

#### Here's what shellcheck currently says:

          ^-- SC3040: In POSIX sh, set option pipefail is undefined.

#### Here's what I wanted or expected to see:

The same.
But the linked page https://github.com/koalaman/shellcheck/wiki/SC3040 should not be empty, but explain something… at least…
